src/main/java/com/hillel/dzhurynskyi/homework/hm12/Box.java

```diff
@@ -0,0 +1,45 @@
+package com.hillel.dzhurynskyi.homework.hm12;
+
+import java.util.Objects;
+
+public class Box<T extends Fruit> {
+    private final T type;
+    private int amount;
```

Comment:
in general, the task is to use generics, so the right way will be using List fruits instead of type + amount and operate with it in all methods